### PR TITLE
Extend supplier_account flow for new details editing flow

### PR DESCRIPTION
**Pull request by Wynndow**
This branch is rebased on this PR: https://github.com/alphagov/digitalmarketplace-functional-tests/pull/483, it's only the last commit which should be reviewed here.

For this trello ticket: https://trello.com/c/QBDqUfVk

We need functional tests to cover the new supplier editing details flow.

This covers hitting the 'Save and confirm' button and checking that certain details are no longer editable. Will need PR's for allowing editing until cofirmation, and for the button, to be merged before these can go in.

### CODE REVIEW

File: features/supplier/supplier_account.feature [hunk 1]
```diff
@@ -7,23 +7,181 @@ Background:
...
+
+  Then I click the summary table 'Change' link for 'VAT number'
+  Then I am on the 'Change your VAT number' page
+  And I see 'Contact cloud_digital@crowncommercial.gov.uk to make changes to your:' text on the pages
+  And I don't see the 'Save and continue' button
+  Then I click the 'Return to company details' link
+
+  Then I click the summary table 'Change' link for 'DUNS number'
+  Then I am on the 'Change your DUNS number' page
+  And I see 'Contact cloud_digital@crowncommercial.gov.uk to make changes to your:' text on the pages
```
Comment (CrystalPea) [inline features/supplier/supplier_account.feature]: DUNS number is not editable even before clicking the button so you could move this step up and merge it with the earlier DUNS number page step.

File: features/supplier/supplier_account.feature [hunk 2]
```diff
@@ -7,23 +7,181 @@ Background:
...
+  Then I see the 'Registration information' summary table filled with:
+    | field                     |  value                                                |
+    | Registered company name   | Toys "ᴙ" Not Us                                       |
+    | Registered company address| 101 Liquidation Street Shutdown N0 TOY United Kingdom |
+    | Registration number       | 18092232                                              |
+    | Trading status            | Other                                                 |
+    | Company size              | Micro                                                 |
+    | VAT number                | Not VAT registered                                    |
+
+  # Can pull the dunsNumber from the supplier and use it with this step
```
Comment (CrystalPea) [inline features/supplier/supplier_account.feature]: Should we remove the comments, or at least some of them, before merging?

File: features/supplier/supplier_account.feature (hunk 1 above)
Comment (Wynndow) [reply features/supplier/supplier_account.feature]: Yep, have moved up. Have also kept here though.

File: features/supplier/supplier_account.feature (hunk 2 above)
Comment (Wynndow) [reply features/supplier/supplier_account.feature]: I'm leaving them in for information. It's quite a long scenario and they make it easier to understand what's going on at certain stages, an to explain some little quirks, I think.

Comment (CrystalPea) [reply features/supplier/supplier_account.feature]: ok 👍

**Comment by Wynndow** [comment]
Hopefully my other PR for updating button content across the supplier app will get merged, and then I can merge the other functional tests PR before this one, maybe with tags to not run on the supplier app until it's out. Then when Kev's button get's merged, same with this PR. If Kev's button goes first I can just undo the rebase and merge this first.

**Comment by CrystalPea** [comment]
Sounds good 👍

**Pull request merged**